tools/build/sprite/player_sprites.py

```diff
@@ -0,0 +1,466 @@
+            )
+
+            if is_back:
+                raster_offset += width * height // 2
+            else:
+                raster_offset += 0x10
+
+    out_bytes += raster_bytes
+    back_out_bytes += raster_bytes_back
+
+    raster_list_offset = len(out_bytes) + 0x10
+    raster_list_offset_back = len(back_out_bytes) + 0x10
+
+    # Raster file offsets
+    raster_offsets_bytes = b""
+    raster_offsets_bytes_back = b""
+    for i in range(len(xml[1])):
+        raster_offsets_bytes += int.to_bytes(raster_list_start + i * 8, 4, "big")
+        raster_offsets_bytes_back += int.to_bytes(
+            raster_list_start_back + i * 8, 4, "big"
+        )
+    raster_offsets_bytes += LIST_END_BYTES
+    raster_offsets_bytes_back += LIST_END_BYTES
+
+    out_bytes += raster_offsets_bytes
+    back_out_bytes += raster_offsets_bytes_back
+
+    # Palette file offsets
+    palette_list_offset = len(out_bytes) + 0x10
+    palette_list_offset_back = len(back_out_bytes) + 0x10
+    palette_offsets_bytes = b""
+    palette_offsets_bytes_back = b""
+    for i, palette_xml in enumerate(xml[0]):
+        palette_offsets_bytes += int.to_bytes(palette_list_start + i * 0x20, 4, "big")
+        front_only = bool(palette_xml.attrib.get("front_only", False))
+        if not front_only:
+            palette_offsets_bytes_back += int.to_bytes(
+                palette_list_start_back + i * 0x20, 4, "big"
+            )
+    palette_offsets_bytes += LIST_END_BYTES
+    palette_offsets_bytes_back += LIST_END_BYTES
+
+    out_bytes += palette_offsets_bytes
+    back_out_bytes += palette_offsets_bytes_back
+
+    header = struct.pack(
+        ">IIII",
+        raster_list_offset,
+        palette_list_offset,
+        max_components,
+        num_variations,
+    )
+    out_bytes = header + out_bytes
+
+    ret = [out_bytes]
+
+    if has_back:
+        back_header = struct.pack(
+            ">IIII",
+            raster_list_offset_back,
+            palette_list_offset_back,
+            0,
+            num_variations,
+        )
+        back_out_bytes = back_header + back_out_bytes
+
+        ret.append(back_out_bytes)
+
+    return ret
+
+
+def build(player_sprite_dir: Path, out_binary: Path) -> None:
+    sprite_order, raster_order = get_orderings_from_xml(player_sprite_dir)
+
+    # Build info
+    with open(player_sprite_dir / "build_info.txt", "r") as f:
+        build_info = f.read()
+
+    # Read all rasters and cache them
+    cur_offset = 0
+    for raster_name in raster_order:
+        # "Weird" raster
+        if (
+            os.path.getsize(player_sprite_dir / "rasters" / f"{raster_name}.png")
+            == 0x10
+        ):
+            special_raster_bytes = (
+                b"\x80\x30\x02\x10\x00\x00\x02\x00\x00\x00\x00\x01\x00\x10\x00\x00"
+            )
+            RASTER_CACHE[raster_name] = PNGInfo(0, 0, cur_offset, special_raster_bytes)
+            cur_offset += 0x10
+            continue
```

Comment:
Have we tried looking at the loading code to figure out what this data is supposed to be? `80300210` looks like it might be an address...

Reply:
8030 doesn't seem like an address to me, but I could be wrong. I think only clover can answer that question and it may have to wait until the code is figured out further. Or he may, already know now. Lol @HailSanta do you?